### PR TITLE
Admin/Volunteer edit account form + Update seed data

### LINKED ISSUE #431: Admin/volunteer modifies profile page
### Summary and Background
Admin/volunteer modifies profile page

### User Story
As an Admin/Volunteer, I want to edit my personal information into the platform, so the Super Admin can contact me and I can highlight my skills.
As a Super Admin, I want to see volunteer/admin's updated personal information, so I can contact/refer to volunteers/admins.

### Solution (Core Elements)
Create admin/volunteer modifies profile page, accessed admin/volunteer navbar. Very similar to account creation #341

Fields:
- First Name*
- Last Name*
- Date of Birth
- Pronouns
- Password*
- Confirm Password*
- Phone Number*
- Emergency Contact Phone Number*
required *

The modifies profile page is a bit different between volunteer and admin. The only difference is that volunteers are asked for skills while admin are not.

Please also ensure fields are validated appropriately according to the designs.

If the fields are valid, clicking the 'Save' button saves the modifications.

After the modal closes, the user is transported back to where they were.

Since we have not implemented profile pictures yet, let the header contain a default profile icon (similar to account creation for now).

### Mockups
Volunteer:

Admin:

Other designs: 
https://www.figma.com/file/tOK6AjkaRZ1NBXo191D64C/Sistering-%2F-User-Authentication-Process?node-id=7%3A3

### Technical Details
- Create a file called `EditAccountForm.tsx`. Use `frontend/src/components/user/AccountForm.tsx` as reference for the UI (the form should be identical)
- Call the `getUserById` resolver to retrieve the user's information and prefill the Formik form with details supplied by the resolver call
- Call the `updateUser` resolver when the save button is clicked to update the user's information

### LINKED ISSUE #498: Update Seed to create employee record for admin user
Currently, in our seed data, our admin user does not have an employee record associated with it. We should update our seed to correct this.

### PR DESCRIPTION
**Pull request by lambo-liu**
## Ticket link

Closes #431 
Closes #498 

## Implementation description
* Refactored `AccountForm` components into separate field components (`TextField` and `SelectorField`)
* Made `AccountForm` dynamic (i.e. components can pass in field data to pre-populate the form => this is used for editing)
* Reworked some of the AccountForm visuals & functions
* Fixed Volunteer/Employee User DTO types to match backend request model
* Created the EditAccountPage, which recycles the `AccountForm` component
* Updated seed data to include admin IDs in the employee table
* Reworked `updateEmployeeUserById()` in the backend to not assume user's role to be volunteer
* Protected `ADMIN_HOMEPAGE` to admin only
* (Sorry for the long change-list! A lot of these are bug fixes to existing code; the main focus of this ticket are in `AccountForm.tsx` and `EditAccountPage.tsx`)

## Steps to test
1. Navigate to `http://localhost:3000/edit-account` with an admin account
2. Try changing some of the fields and submitting the form, the fields should also be updated in backend (NOTE: DOB and Pronouns are not stored for admins, this is a TODO).
3. Repeat this for a volunteer account. Ensure that selectors for skills and languages work as intended.

## What should reviewers focus on?
* Functionality, didn't break any existing code

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/pages/EditAccountPage.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+  mutation UpdateVolunteerUser($id: ID!, $volunteer: UpdateVolunteerUserDTO!) {
+    updateVolunteerUserById(id: $id, volunteerUser: $volunteer) {
+      id
+    }
+  }
+`;
+const EditAccountPage = (): React.ReactElement => {
+  const { authenticatedUser } = useContext(AuthContext);
+
+  const [key, setKey] = useState<number>(0); // Used to force a re-render
```
Comment (MatoPlus) [inline frontend/src/components/pages/EditAccountPage.tsx:82]: interesting 🤔

File: frontend/src/components/pages/NewAccountPage.tsx [hunk 2]
```diff
@@ -73,7 +74,9 @@ const NewAccountPage = (): React.ReactElement => {
         />
         <Divider my={8} />
         <AccountForm
+          mode={AccountFormMode.CREATE}
           isAdmin={isAdmin}
+          email="testemail123@domain.com"
```
Comment (MatoPlus) [inline frontend/src/components/pages/NewAccountPage.tsx:79]: Let's add a TODO comment here to replace this

Comment (MatoPlus) [reply frontend/src/components/pages/NewAccountPage.tsx:79]: also @eevanwong fyi

File: backend/services/implementations/userService.ts [hunk 3]
```diff
@@ -1147,43 +1147,42 @@ class UserService implements IUserService {
...
+          user: {
+            update: {
+              firstName: employeeUser.firstName,
+              lastName: employeeUser.lastName,
+              role: oldEmployeeUser?.user.role,
+              phoneNumber: employeeUser.phoneNumber,
+              emergencyContactName: employeeUser.emergencyContactName,
+              emergencyContactPhone: employeeUser.emergencyContactPhone,
+              emergencyContactEmail: employeeUser.emergencyContactEmail,
+              languages: employeeUser.languages,
```
Comment (MatoPlus) [inline backend/services/implementations/userService.ts:1173]: was there any reason why we decided to extract this?

Comment (lambo-liu) [reply backend/services/implementations/userService.ts:1173]: I needed to use the oldEmployeeUser's role in the update - not sure how else to get the original role of the user

Comment (lambo-liu) [reply backend/services/implementations/userService.ts:1173]: I don't think removing the transaction would matter too much here anyway?

Comment (MatoPlus) [reply backend/services/implementations/userService.ts:1173]: i see. Yea I also think it's fine in this scenario

File: frontend/src/components/user/AccountForm.tsx [hunk 4]
```diff
@@ -334,32 +422,8 @@ const AccountForm = ({
...
-                        phoneNumber: values.phoneNumber,
-                        hireDate: moment(new Date()).format("YYYY-MM-DD"),
-                        dateOfBirth: values.dateOfBirth,
-                        skills: values.skills.map((skill) => skill.id),
-                        branches: [],
-                      })
-              }
             >
-              Create
+              {mode === AccountFormMode.CREATE ? "Create" : "Edit"}
```
Comment (MatoPlus) [inline frontend/src/components/user/AccountForm.tsx:426]: I don't think this quite matches design but we can follow up.
image

**Review by MatoPlus — APPROVED** [review]
🐐

**Pull request merged**